TDL15-CloudFormation-Jenkins/cf-deploy-jenkins.yaml

```diff
@@ -0,0 +1,245 @@
+            sqlite-devel tk-devel xz-devel zlib-devel libffi-devel
+          sudo wget https://www.python.org/ftp/python/${PythonVersion}/Python-${PythonVersion}.tgz
+          tar -xf Python-${PythonVersion}.tgz
+          cd Python-${PythonVersion}
+          ./configure --enable-optimizations # Setup installation by running the configure script, performs a number of checks to make sure all of the dependencies on your system are present
+          make -j 4 # Start the Python 3.8 build process 
+          sudo make altinstall #install the Python binaries
+          alias python="/usr/local/bin/python3.8"
+          # intall boto3
+          pip3.8 install boto3
+          # install terraform 
+          sudo wget https://releases.hashicorp.com/terraform/${TerraformVersion}/terraform_${TerraformVersion}_linux_amd64.zip
+          unzip terraform_${TerraformVersion}_linux_amd64.zip && sudo rm terraform_${TerraformVersion}_linux_amd64.zip -f && sudo mv terraform /usr/local/bin/
+
+  # Define the Security Group that will be appended to the ENI of the Instance.  
+  ServerSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: Security Group that will be used for the Jenkins instance. Open ports 22, and the JenkinsPort
+      VpcId: !Ref VpcId
+      SecurityGroupIngress:
+        - IpProtocol: tcp
+          FromPort: 22
+          ToPort: 22
+          CidrIp: !Ref SSHLocation
+        - IpProtocol: tcp
+          FromPort: !Ref JenkinsPort
+          ToPort: !Ref JenkinsPort
+          CidrIp: !Ref SSHLocation       
+
+  # Create the EBS volume that we will use for /var/lib/jenkins
+  JenkinsDataVolume:
+    Type: AWS::EC2::Volume
+    Properties:
+      Size: !Ref VolumeSize
+      SnapshotId:  !If [UseSnapshot, !Ref SnapshotId, !Ref "AWS::NoValue"]
+      Encrypted: true
+      AvailabilityZone: !GetAtt JenkinsInstance.AvailabilityZone
+      Tags:
+        - Key: Role
+          Value: Data
+    DeletionPolicy: Snapshot
+
+  # Attach EBS volume to instance
+  MountPoint:
+    Type: AWS::EC2::VolumeAttachment
+    Properties:
+      InstanceId: !Ref JenkinsInstance
+      VolumeId: !Ref JenkinsDataVolume
+      Device: /dev/sdb
+
+# Specify any outputs for the stack.
+Outputs:
+  TemplateID:
+    Description: 'Jenkins Server Template'
+    Value: 'Jenkins Server Template'
+  PublicIp:
+    Description: 'The Public IP address of the EC2 Instance'
+    Value: !GetAtt JenkinsInstance.PublicIp
+  ServerAddress:
+    Description: 'The public DNS address for this instance'
+    Value: !GetAtt JenkinsInstance.PublicDnsName
+  JenkinsPass:
+    Description: Jenkins One Time Install Password 
+    Value: !Sub "ssh ec2-user@${JenkinsInstance.PublicIp} sudo cat /var/lib/jenkins/secrets/initialAdminPassword"
+  JenkinsKey:
+    Description: Jenkins SSH Public Key
+    Value: !Sub "ssh ec2-user@${JenkinsInstance.PublicIp} sudo cat /var/lib/jenkins/.ssh/id_rsa.pub"
```

Comment:
same here,
Who is going use the public key?

Reply:
For further integrate with GitHub.

Reply:
Ok, I delete this two outputs.